torch/_inductor/codegen/triton.py

```diff
@@ -1138,6 +1181,140 @@ def indexing(
                     mask_vars,  # type: ignore[arg-type]
                 )
 
+            def match_mod_div_block() -> Union[BlockPtrOptions, None]:
+                """
+                Matches higher-dimensional blocks coming from FloorDiv and ModularIndexing.
+
+                Example expression to match:
+                   sN * ((xindex//(d1 * ... * d(N-1))))
+                       + s1 * ModularIndexing(xindex, 1, d1)
+                       + ...
+                       + s(N-1) * ModularIndexing(xindex, d(N-2), d(N-1))
+                       + offset
+
+                This iterates over a block of shape (dN, ..., d1) and stride (sN, ..., s1).
+                """
+
+                # Get info about iteration ranges
+                # Check the index variable. We expect a linear index.
+                if len(symbols) != 1:
+                    return None
+                index_var = symbols[0]
+
+                # Get the dims in reverse numerical order. E.g. x(N-1), ..., x0.
+                # This matches the order of dims in torch tensors and triton blocks.
+                sizevars = V.graph.sizevars
+                var_ranges = next(iter(self.range_tree_nodes.values())).var_ranges
+                dims = dict(
+                    reversed(
+                        [
+                            (var, sizevars.lookup_precomputed_size(size))
+                            for (var, size) in var_ranges.items()
+                        ]
+                    )
```

Comment:
I don't think the order here is guaranteed.  You might also have unrelated sizes in the range.  The test case to try is a single kernel with two different indexings in it, so this could pull in more symbols than are in the expr.

You might want to filter things by expr.free_symbols then sort by V.graph.sizevars.size_hint.